builtins/amp-pixel.js

```diff
@@ -16,59 +16,81 @@
 
 import {BaseElement} from '../src/base-element';
 import {Layout} from '../src/layout';
-import {urlReplacementsForDoc} from '../src/url-replacements';
+import {dev, user} from '../src/log';
 import {registerElement} from '../src/custom-element';
 import {toggle} from '../src/style';
-import {user} from '../src/log';
+import {urlReplacementsForDoc} from '../src/url-replacements';
+import {viewerForDoc} from '../src/viewer';
+
+const TAG = 'amp-pixel';
+
 
 /**
- * @param {!Window} win Destination window for the new element.
- * @this {undefined}  // Make linter happy
- * @return {undefined}
+ * A simple analytics instrument. Fires as an impression signal.
  */
-export function installPixel(win) {
-  class AmpPixel extends BaseElement {
+export class AmpPixel extends BaseElement {
+
+  /** @override */
+  constructor(element) {
+    super(element);
 
-    /** @override */
-    getPriority() {
-      // Loads after other content.
-      return 1;
-    }
+    /** @private {?Promise} */
+    this.triggerPromise_ = null;
+  }
 
-    /** @override */
-    isLayoutSupported(layout) {
-      return layout == Layout.FIXED;
-    }
+  /** @override */
+  isLayoutSupported(layout) {
+    // "fixed" layout is supported, but in reality it's always width/height = 0.
+    return layout == Layout.FIXED;
```

Comment:
This is the layout that's automatically calculated for all `amp-pixel` right now. Basically, all `amp-pixel` are "layout=fixed" unless explicitly overridden. And, of course, no one explicitly overrides that. Thus, here we have no choice but allow it. Alternatively, we'd get bunch of errors.